handler/handler.go

```diff
@@ -12,23 +13,34 @@ type CepHandler struct {
 	CepApis      map[string]api.Api
 }
 
-func (h *CepHandler) Handle(ctx *gin.Context) {
-	cep := ctx.Param("cep")
-	ctx.Header("Content-Type", "application/json; charset=utf-8")
+type responseError struct {
+	Error string `json:"error"`
+}
+
+func renderJSON(w http.ResponseWriter, code int, data interface{}) {
+	j, _ := json.Marshal(data)
+	w.WriteHeader(code)
+	w.Write(j)
+}
+
+func (h *CepHandler) Handle(w http.ResponseWriter, r *http.Request) {
+	cep := strings.Split(r.URL.Path, "/")[1]
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
 
 	preferredApi, ok := h.CepApis[h.PreferredApi]
 	if !ok {
-		ctx.JSON(http.StatusInternalServerError, gin.H{"error": "preferred api not available"})
+		renderJSON(w, http.StatusInternalServerError, &responseError{Error: "preferred api not available"})
```

Comment:
you can omit the `Error:` when instantiating the struct here, so `&responseError{"preferred api not available"}`